crates/rspack_fs/src/fs.rs

```diff
@@ -0,0 +1,10 @@
+use crate::{
+  AsyncFileSystem, AsyncReadableFileSystem, AsyncWritableFileSystem, SyncFileSystem,
+  SyncReadableFileSystem, SyncWritableFileSystem,
+};
+
+pub trait FileSystem: AsyncFileSystem + SyncFileSystem {}
+
+pub trait ReadableFileSystem: AsyncReadableFileSystem + SyncReadableFileSystem {}
```

Comment:
There doesn't seem to be anywhere to use `ReadableFileSystem` and `WritableFileSystem`, maybe we can
``` rust
trait FileSystem: AsyncReadableFileSystem + SyncReadableFileSystem + AsyncWritableFileSystem + SyncWritableFileSystem {}

impl<T> FileSystem for T where T: AsyncReadableFileSystem + SyncReadableFileSystem + AsyncWritableFileSystem + SyncWritableFileSystem {}
```

Reply:
I consider remove `AsyncReadableFileSystem` and `SyncReadableFileSystem` and only keep `ReadableFileSystem` in the future